### PR TITLE
Added a way to configure StatsdClient to send batch of metrics in one

### PR DESCRIPTION
**Pull request by Unix4ever**
udp packet.

It now accepts two parameters:
- `maxPacketSize` limit packet size, so when we reach that value, packet
  will be sent immediately.
- `flushInterval` flush metrics by the intervals defined in this
  parameter.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -179,3 +220,42 @@ func (c *StatsdClient) SendEvent(e event.Event) error {
...
+	}
+}
+
+func (c *StatsdClient) flushStatBuffer() bool {
+	if c.buffer.Len() == 0 {
+		return false
+	}
+	_, err := c.conn.Write(c.buffer.Bytes())
+	if err == nil {
+		c.buffer.Reset()
```
Comment (smira) [inline client.go]: we would overflow buffer is we don't reset it? does it make sense to ignore errors here and reset always?

File: client.go [hunk 2]
```diff
@@ -161,7 +178,31 @@ func (c *StatsdClient) send(stat string, format string, value interface{}) error
...
+
+		if c.buffer.Len() + len(format) > c.maxPacketSize {
+			c.flushStatBuffer()
+			c.Timer.Reset(c.flushInterval)
+		}
+
+		if c.buffer.Len() == 0 {
+			c.buffer.Write([]byte(metricValue))
+		} else {
+			fmt.Fprintf(&c.buffer, "\n%s", metricValue)
```
Comment (smira) [inline client.go]: `c.buffer.Write` should be really effective... 

can we re-write it into:

```
 if c.buffer.Len() > 0 {
     c.buffer.WriteByte('\n')
 }

c.buffer.WriteString(metricValue)
```

File: client.go [hunk 3]
```diff
@@ -161,7 +178,31 @@ func (c *StatsdClient) send(stat string, format string, value interface{}) error
...
+	metricValue := fmt.Sprintf(format, value)
+
+	var err error
+
+	if c.flushInterval != 0 {
+		if !c.timerRunning {
+			c.startTimer()
+		}
+
+		c.statsMutex.Lock()
```
Comment (smira) [inline client.go]: can we do `defer statsMutex.Unlock()` just in case of panic? (instead of Unlock() down below)

File: client.go (hunk 2 above)
Comment (Unix4ever) [inline client.go]: oh, really. That's much simpler :)

**Pull request merged**